### ISSUE TITLE
New release

### ISSUE THREAD

**Issue by iamsmooth** [aeonix/aeon#257]
Is this a good point for a new release? There seem to be a lot of upstream merges as well as the much needed update to the seed nodes.

Or is there some ongoing work that would make it better to wait?

**Comment by BigslimVdub** [comment]
#255 merge then seems good for a new point release. 

Is there a consensus change? I do not recall anything that would require this being added since prior point release. If not then looks good for a point release. It has been long enough and lots of enhancements plus compile fixes.

**Comment by stoffu** [comment]
I agree that a new point release is indeed desired now because of the stale inactive seed nodes issue (eg https://www.reddit.com/r/Aeon/comments/r24ubc/comment/hm54144/).

**Comment by iamsmooth** [comment]
Are the upstream PRs ready to merge (both CLI and GUI)?

**Comment by ghost** [comment]
Should work well with new seed nodes.

**Comment by iamsmooth** [comment]
Bump 

> Are the upstream PRs ready to merge (both CLI and GUI)?

**Comment by stoffu** [comment]
Sorry for my slow response, been very busy these days. They should be ready, but please check by yourself if you can.

**Comment by ghost** [comment]
Reviewed

**Comment by BigslimVdub** [comment]
Looks like GUI has one more PR plus a version bump. https://github.com/aeonix/aeon-gui/pull/58

Same as CLI

**Comment by iamsmooth** [comment]
This looks close to ready. There are a couple of MacOS PRs that need review. They look small, but I'm not familiar enough with what they are doing to review myself.

After that we will need version bump and binaries built.

**Comment by BigslimVdub** [comment]
Recommend version bump commit and release name poll on social media. #261 can stay open for the time being. Same for gui pr.

**Comment by stoffu** [comment]
There's no need for release name poll because this is going to be just a point release `v0.14.2.0-aeon`.

**Comment by iamsmooth** [comment]
@stoffu Are you able to do a version bump commit and also please explain how the new process for binaries works. Thanks.

**Comment by stoffu** [comment]
Currently I’m struggling with annoying issues in the GUI (crashes when opening the Settings tab in Windows & Mac, not launching at all on Linux). I’m configuring the GitHub Actions CI such that a custom built Qt can be used. When completed, obtaining release binaries should be as simple as just downloading the built artifacts from the CI. For example, you can get the current master binaries from https://github.com/aeonix/aeon-gui/actions/runs/1799574070#artifacts (which don't run correctly due to the Qt issues).

The issue is relevant for the GUI only, so we could make new release for the CLI first, and do the GUI release later when the Qt issue gets solved. Should we go this way, or is it better to release the CLI & GUI at the same time?

**Comment by iamsmooth** [comment]
Let's wait a little while and see if you can make any progress with the GUI. If not we can push out the CLI separately.

**Comment by stoffu** [comment]
Finally I think I figured out the right way of producing release binaries. With these patches https://github.com/aeonix/aeon/pull/264 & https://github.com/aeonix/aeon-gui/pull/63, the following binaries have been generated by GitHub Actions CI (the corresponding CI runs are https://github.com/aeonix/aeon/actions/runs/1896600423 & https://github.com/stoffu/monero-core/actions/runs/1916874316/attempts/6, respectively):

Please download the following release candidate binaries and report if you see any issues:

- Linux CLI: https://drive.google.com/file/d/1E52hKWVuSFj4FdDgUW507hs7MRWxPBM0/view?usp=sharing
- Mac CLI: https://drive.google.com/file/d/1afhDBsQr3mckJTh4YVI5Ep398dvyvxjW/view?usp=sharing
- Windows CLI: https://drive.google.com/file/d/1lrzlWwjYYjvkr1jSccl9Pzzg4YOLoYJ7/view?usp=sharing
- Linux GUI: https://drive.google.com/file/d/14f0JWMTfnJUqloTUE7pfaeMp12Uce7WK/view?usp=sharing
- Mac GUI: https://drive.google.com/file/d/1gwRnlEVFqLgpy5iVC54qoqb0fIrn-FkI/view?usp=sharing
- Windows GUI: https://drive.google.com/file/d/1lrzlWwjYYjvkr1jSccl9Pzzg4YOLoYJ7/view?usp=sharing

SHA256 hashes:
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

SHA256(aeon-linux-x64-v0.14.1.0-aeon-125-gf5bbda63.tar.bz2)= 791b572ca6e013c5cf315cbe12f6d75890157e5f858198517036e8e545b0ede4
SHA256(aeon-mac-x64-v0.14.1.0-aeon-125-gf5bbda63.tar.bz2)= 6ce04815a17e5f4cb115ded2d7676f4a438ae11a6b4f1241c95c849cdc03a169
SHA256(aeon-win-x64-v0.14.1.0-aeon-125-gf5bbda63.zip)= f90e6ec05a12b211cd122fe6706a6590add0feba404baca902957fcb8edb3c88
SHA256(aeon-gui-linux-x64-v0.14.1.0-aeon-71-gf322f125.tar.bz2)= c41be3ad3be6d7fe64de9d34e7e5c0b657eb819351182d2f01cdcb09c763ab3c
SHA256(aeon-gui-mac-x64-v0.14.1.0-aeon-71-gf322f125.tar.bz2)= 41db747333e2d57b68aab1b7b3cbdd7198739a8eb203524dc1bca32914966cfa
SHA256(aeon-gui-win-x64-v0.14.1.0-aeon-71-gf322f125.zip)= 1dfee16f6454f1281eb5f56ea552d238ffff4599accaad331903e73405128dc8
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEEhJ/ELagGfXubAPJKQdq4NDqewBIFAmIe4MMACgkQQdq4NDqe
wBKitw/+IGRpYqj4o33DzjwIE3tkmyx2SaiHGugK0y3+ItPxJDP++ORndOOqXlCZ
flYmQ0ivpJWqdK/47YLwSIUm2oSeEkUryU1AlJBJV3F7Tnx3W+yHCXYJOEhRfTYX
AIThdIP9/2gj+zGLmSg5v81FI64Nil6im9Oo35/3+06FXKIrUrgtrdsNd1WcTPiY
QVLj4Z6mLwzKyfq7ijkSHvmFDJZZX2y+mhIoS/O/ck5vNaisZYZaKN1g1ZTV0RHJ
DGoY5W2dyJo9ERtMRIktBYpspZCewdMSzxSEO56NEU+dL5/DTMOdz3LCNW1P4Jbg
QgPK1wci+t1UPKhJgF2v4hdJm/wkw47koZZq68JaRzos75Lma21V6vRN6cW4d6k7
TrVn+1DvgKF5QJ6jIxA9rbtw+CwQETz6oWRg30M4eDdK5HsF8S63VBG615d2fWSL
ug+s+XSHP7fwS0otDDzASUgHm2dV2dKTZhcAvlysA8EDVLBiE+2+2E1vpR9y0LrR
pMfqlr+xSvEi4earJ+pXUQw+jjjP3+jiGyUU+naSvLqZb0jmnZ4tckdfqCQ+jaEi
92dOqTKmzZzA+YJKw9JxlWOH6e7PEILfYTuzaQwLDAB000uVgYpMEkTMlsnKm/Un
xg0jy+pVwLXjtBZHeV1EMAHIvNE4tvlZ2uOY78+9xZK45sKN/Aw=
=6AdH
-----END PGP SIGNATURE-----
```

#### Some explanations

- For some unknown reason, the GUI suffers from some annoying crashing issues as noted above when built with Qt 5.15.2 which is the default available version from package managers on all 3 platforms (mac/linux/win). I looked for ways of building it with other version of Qt, and found this nice utility https://github.com/jurplel/install-qt-action for easily installing various version of Qt within GitHub Action. I found Qt 5.14.2 to be just fine.

- There was a tricky issue in deploying the GUI on Linux: in Monero, the release GUI binaries for Linux have been built using statically-built Qt, so the deployment script `linuxdeploy_helper.sh` does not copy any `*.so` of Qt. This is problematic for using the above-mentioned Qt installer because the installed Qt was built as shared library. In order to deploy the Linux binary correctly, I chose to use https://github.com/probonopd/linuxdeployqt and bundle everything into a single file `aeon-wallet-gui.AppImage`.

**Comment by BigslimVdub** [comment]
> Finally I think I figured out the right way of producing release binaries. With these patches #264 & [aeonix/aeon-gui#63](https://github.com/aeonix/aeon-gui/pull/63), the following binaries have been generated by GitHub Actions CI (the corresponding CI runs are https://github.com/aeonix/aeon/actions/runs/1896600423 & https://github.com/stoffu/monero-core/actions/runs/1916874316/attempts/6, respectively):
> 
> Please download the following release candidate binaries and report if you see any issues:
> 
> * Linux CLI: https://drive.google.com/file/d/1E52hKWVuSFj4FdDgUW507hs7MRWxPBM0/view?usp=sharing
> * Mac CLI: https://drive.google.com/file/d/1afhDBsQr3mckJTh4YVI5Ep398dvyvxjW/view?usp=sharing
> * Windows CLI: https://drive.google.com/file/d/1lrzlWwjYYjvkr1jSccl9Pzzg4YOLoYJ7/view?usp=sharing
> * Linux GUI: https://drive.google.com/file/d/14f0JWMTfnJUqloTUE7pfaeMp12Uce7WK/view?usp=sharing
> * Mac GUI: https://drive.google.com/file/d/1gwRnlEVFqLgpy5iVC54qoqb0fIrn-FkI/view?usp=sharing
> * Windows GUI: https://drive.google.com/file/d/1lrzlWwjYYjvkr1jSccl9Pzzg4YOLoYJ7/view?usp=sharing
> 
> SHA256 hashes:
> 
> ```
> -----BEGIN PGP SIGNED MESSAGE-----
> Hash: SHA256
> 
> SHA256(aeon-linux-x64-v0.14.1.0-aeon-125-gf5bbda63.tar.bz2)= 791b572ca6e013c5cf315cbe12f6d75890157e5f858198517036e8e545b0ede4
> SHA256(aeon-mac-x64-v0.14.1.0-aeon-125-gf5bbda63.tar.bz2)= 6ce04815a17e5f4cb115ded2d7676f4a438ae11a6b4f1241c95c849cdc03a169
> SHA256(aeon-win-x64-v0.14.1.0-aeon-125-gf5bbda63.zip)= f90e6ec05a12b211cd122fe6706a6590add0feba404baca902957fcb8edb3c88
> SHA256(aeon-gui-linux-x64-v0.14.1.0-aeon-71-gf322f125.tar.bz2)= c41be3ad3be6d7fe64de9d34e7e5c0b657eb819351182d2f01cdcb09c763ab3c
> SHA256(aeon-gui-mac-x64-v0.14.1.0-aeon-71-gf322f125.tar.bz2)= 41db747333e2d57b68aab1b7b3cbdd7198739a8eb203524dc1bca32914966cfa
> SHA256(aeon-gui-win-x64-v0.14.1.0-aeon-71-gf322f125.zip)= 1dfee16f6454f1281eb5f56ea552d238ffff4599accaad331903e73405128dc8
> -----BEGIN PGP SIGNATURE-----
> 
> iQIzBAEBCAAdFiEEhJ/ELagGfXubAPJKQdq4NDqewBIFAmIe4MMACgkQQdq4NDqe
> wBKitw/+IGRpYqj4o33DzjwIE3tkmyx2SaiHGugK0y3+ItPxJDP++ORndOOqXlCZ
> flYmQ0ivpJWqdK/47YLwSIUm2oSeEkUryU1AlJBJV3F7Tnx3W+yHCXYJOEhRfTYX
> AIThdIP9/2gj+zGLmSg5v81FI64Nil6im9Oo35/3+06FXKIrUrgtrdsNd1WcTPiY
> QVLj4Z6mLwzKyfq7ijkSHvmFDJZZX2y+mhIoS/O/ck5vNaisZYZaKN1g1ZTV0RHJ
> DGoY5W2dyJo9ERtMRIktBYpspZCewdMSzxSEO56NEU+dL5/DTMOdz3LCNW1P4Jbg
> QgPK1wci+t1UPKhJgF2v4hdJm/wkw47koZZq68JaRzos75Lma21V6vRN6cW4d6k7
> TrVn+1DvgKF5QJ6jIxA9rbtw+CwQETz6oWRg30M4eDdK5HsF8S63VBG615d2fWSL
> ug+s+XSHP7fwS0otDDzASUgHm2dV2dKTZhcAvlysA8EDVLBiE+2+2E1vpR9y0LrR
> pMfqlr+xSvEi4earJ+pXUQw+jjjP3+jiGyUU+naSvLqZb0jmnZ4tckdfqCQ+jaEi
> 92dOqTKmzZzA+YJKw9JxlWOH6e7PEILfYTuzaQwLDAB000uVgYpMEkTMlsnKm/Un
> xg0jy+pVwLXjtBZHeV1EMAHIvNE4tvlZ2uOY78+9xZK45sKN/Aw=
> =6AdH
> -----END PGP SIGNATURE-----
> ```
> 
> #### Some explanations
> * For some unknown reason, the GUI suffers from some annoying crashing issues as noted above when built with Qt 5.15.2 which is the default available version from package managers on all 3 platforms (mac/linux/win). I looked for ways of building it with other version of Qt, and found this nice utility https://github.com/jurplel/install-qt-action for easily installing various version of Qt within GitHub Action. I found Qt 5.14.2 to be just fine.
> * There was a tricky issue in deploying the GUI on Linux: in Monero, the release GUI binaries for Linux have been built using statically-built Qt, so the deployment script `linuxdeploy_helper.sh` does not copy any `*.so` of Qt. This is problematic for using the above-mentioned Qt installer because the installed Qt was built as shared library. In order to deploy the Linux binary correctly, I chose to use https://github.com/probonopd/linuxdeployqt and bundle everything into a single file `aeon-wallet-gui.AppImage`.

Do the above binaries have all of the expected changes set for v0.14.2.0 (so far) and are pretty much pre-release binaries just versioned v0.14.1.0?

**Comment by stoffu** [comment]
@BigslimVdub Yes.

**Comment by BigslimVdub** [comment]
Ready for version bump PR?

**Comment by stoffu** [comment]
I’m a bit reluctant to do that since currently there does seem to exist an issue of aeond occasionally getting stuck while in initial blockchain download, as was complained in #265 and confirmed in #267. I’m currently trying to figure out what’s going on, but my investigation is slow due to my other life duties. I hope to get this fixed in a few weeks or so.

Or should we rush the release now anyway ignoring this issue?

**Comment by BigslimVdub** [comment]
Honestly I’ve had intermittent initial chain download issues since 2016 but they have been a lot less since rebase and it may also be related to outdated seed nodes since closing daemon and restarting usually resolves the issue if it’s stuck or can’t start initial sync. 

Are you still seeing issues with a current master pull with the updated seed node addresses? I had no issues on OS X m1 and Ubuntu 20.04 with full downloads from 0. I did not try windows or intel OS X.

**Comment by stoffu** [comment]
Yes, I did observe this IBD stuck issue on macOS and Ubuntu. This can be confirmed also by running the GitHub CI of #267 yourself.

**Comment by iamsmooth** [comment]
Seems fine to continue working on the problem unless unable to make progress in which case it may be worthwhile to get the other improvements out.

**Comment by stoffu** [comment]
The IBD test in #267 now passes, so I think we can move on to releasing.

**Comment by iamsmooth** [comment]
Is there a way to get CI to upload binaries to the release automatically or does it need to be done manually?

**Comment by stoffu** [comment]
> Is there a way to get CI to upload binaries to the release automatically or does it need to be done manually?

I quickly searched the web and found this https://github.com/softprops/action-gh-release, so I'm trying it out in #272.
Please merge it along with #271 (which fixes a build issue on Windows), and then create another release with tag `v0.14.2.1-aeon`. Hopefully the binaries will be pushed to the Release page automatically.

**Comment by iamsmooth** [comment]
No sign of binaries. I guess it didn't work.

**Comment by stoffu** [comment]
It did work. All the binaries have been uploaded there.

**Comment by iamsmooth** [comment]
Indeed. I guess there is some processing delay.

**Comment by iamsmooth** [comment]
Can we upload the GUI binaries in a similar manner?

**Comment by stoffu** [comment]
Yes, once merged https://github.com/aeonix/aeon-gui/pull/64 and created a release with tag `v0.14.2.1-aeon`, the binaries will be auto uploaded.

**Comment by iamsmooth** [comment]
Does it make sense to sign hashes on these? Someone asked on reddit. I'm not sure personally whether it makes sense but I guess it guards against the repo being compromised and the binaries silently replaced?

**Comment by stoffu** [comment]
I plan to sign on the hashes. Just I haven't checked if they all run fine. Will put the signature here soon.
